lib/elixir/src/elixir_exp_clauses.erl

```diff
@@ -81,6 +81,10 @@ do_cond(Meta, {Key, _}, _Acc, E) ->
 'receive'(Meta, KV, E) when not is_list(KV) ->
   compile_error(Meta, ?m(E, file), "invalid arguments for receive");
 'receive'(Meta, KV, E) ->
+  RaiseError = fun(Kind) ->
+    compile_error(Meta, ?m(E, file), "duplicated ~ts clauses given for receive", [Kind])
+  end,
+  ok = assert_at_most_once('after', KV, 0, RaiseError),
```

Comment:
We also need to do the same for `do`. We could write it like this: `case x, do: (1 -> y), do: (2 -> z)`